giraffe/src/transforms/line.ts

```diff
@@ -1,13 +1,30 @@
-import {Table, LineLayerSpec, LineData} from '../types'
+import {Table, LineLayerSpec, LineData, LinePosition} from '../types'
 import {FILL} from '../constants/columnKeys'
 import {createGroupIDColumn, getNominalColorScale} from './'
 
+const getPrevCumulativeDomainValue = (
+  lineData: LineData,
+  groupID: number,
+  domainLabel: string
+): number => {
+  if (groupID < 1) {
+    return 0
+  }
+  const currentGroupLength = lineData[groupID][domainLabel].length
+  const previousGroupLength = lineData[groupID - 1][domainLabel].length
+  if (previousGroupLength > currentGroupLength) {
+    return lineData[groupID - 1][domainLabel][currentGroupLength]
+  }
```

Comment:
this block has a couple of potential undefined accesses (`cannot access property 'length' of null`). not sure how certain we can be that this data exists, but if we want to be more guarded we could do something like:
```ts
const currentGroup = lineData[groupID]
const previousGroup = lineData[groupID - 1]

if (!currentGroup || !previousGroup) {
  return 0
}

const currentDomainLabel = currentGroup[domainLabel]
const previousDomainLabel = previousGroup[domainLabel]

if (currentDomainLabel && previousDomainLabel) {
  if (previousDomainLabel.length > currentDomainLabel.length) {
    return previousDomainLabel[currentDomainLabel.length]
  }
}

return 0
```

alternatively, you could be kinda lazy and yolo it (not a fan of this):

```ts
try {
  const currentGroupLength = lineData[groupID][domainLabel].length
  const previousGroupLength = lineData[groupID - 1][domainLabel].length
  if (previousGroupLength > currentGroupLength) {
    return lineData[groupID - 1][domainLabel][currentGroupLength]
  }
} catch () {
  // chomp
  return 0
}
```

Reply:
Yes, I thought a while on this block. `return 0` happening twice did not feel ideal to me and i would like to come up with something more elegant here. I will look into your suggestion as I write the tests.